AirCasting/Dashboard/Reordering/CustomSwipeAction.swift

```diff
@@ -0,0 +1,79 @@
+// Created by Lunar on 29/08/2022.
+//
+
+import Foundation
+import SwiftUI
+
+// Adds custom swipe action to a given view
+struct SwipeActionView: ViewModifier {
+    let trailingAction: () -> Void
+    
+    @State private var offset: CGFloat = 0
+    @State private var prevOffset: CGFloat = 0
+    
+    init(trailingAction: @escaping () -> Void) {
+        self.trailingAction = trailingAction
+    }
+    
+    func body(content: Content) -> some View {
+        content
+            .offset(x: (offset > 0) ? 0 : offset)
+        // animate the view as `offset` changes
+            .animation(.spring(), value: offset)
+        // allows the DragGesture to work even if there are now interactable
+        // views in the row
+            .contentShape(Rectangle())
+        // The DragGesture distates the swipe. The minimumDistance is there to
+        // prevent the gesture from interfering with List vertical scrolling.
+            .gesture(DragGesture(minimumDistance: 10,
+                                 coordinateSpace: .local)
+                .onChanged { gesture in
+                    offset = gesture.translation.width
+                }
+                .onEnded { _ in
+                    if !checkAndHandleFullSwipe(for: trailingAction, edge: .trailing, width: -UIScreen.main.bounds.size.width) {
+                        offset = 0
+                    }
+                    prevOffset = offset
+                })
```

Comment:
Indentation seems weird here, I know XCode likes it but just pointing it out

Reply:
1/2 I think you can use a shorter form here, please try to use this code here and on 2/2 part of the one that I will provide, together. This is just a suggestion but I think it can be cleaner:
```
 .onEnded { _ in
                    checkAndHandleFullSwipe(for: trailingAction, edge: .trailing, width: -UIScreen.main.bounds.size.width)
                    prevOffset = offset
                })
```